src/server/routes/backend-wallet/signMessage.ts

```diff
@@ -39,22 +42,22 @@ export async function signMessage(fastify: FastifyInstance) {
       const { "x-backend-wallet-address": walletAddress } =
         request.headers as Static<typeof walletHeaderSchema>;
 
-      const wallet = await getWallet({
-        chainId: 1,
-        walletAddress,
-      });
-
-      const signer = await wallet.getSigner();
-
-      let signedMessage;
-      if (isBytes) {
-        signedMessage = await signer.signMessage(
-          ethers.utils.arrayify(message),
+      if (isBytes && !isHex(message)) {
+        throw createCustomError(
+          '"isBytes" is true but message is not hex.',
+          StatusCodes.BAD_REQUEST,
+          "INVALID_MESSAGE",
         );
-      } else {
-        signedMessage = await signer.signMessage(message);
       }
 
+      const account = await getAccount({
+        chainId: 1,
+        from: getChecksumAddress(walletAddress),
+      });
+      const signedMessage = await account.signMessage({
+        message: isBytes ? fromHex(message as Hex, "string") : message,
```

Comment:
if the message is raw bytes/hex, you need to pass it like this:

`account.signMessage({ message: { raw: myBytes }}`

in your test you can quickly write a sign message with the SDK and compare the output, it will be different. Got bitten by this in the past, make sure that raw bytes is passed in that `raw` prop